explainers/security-privacy-questionnaire-post-task.md

```diff
@@ -0,0 +1,137 @@
+# `scheduler.postTask()`: Security and Privacy Questionnaire Answers
+
+The following are the answers to the W3C TAG's
+[security and privacy self-review questionnaire](https://w3ctag.github.io/security-questionnaire/).
+
+> 01. What information might this feature expose to Web sites or other parties,
+>     and for what purposes is that exposure necessary?
+
+The features do not directly expose any information. Indirectly, consumers of
+the `postTask()` API can measure information about tasks they schedule, such as
+queueing time. In Q9, we discuss the potential for side channel attacks using
+this information in the cross-origin case.
+
+> 02. Do features in your specification expose the minimum amount of information
+>     necessary to enable their intended uses?
+
+Yes.
+
+> 03. How do the features in your specification deal with personal information,
+>     personally-identifiable information (PII), or information derived from
+>     them?
+
+They do not consume such information.
+
+> 04. How do the features in your specification deal with sensitive information?
+
+They do not consume such information.
+
+> 05. Do the features in your specification introduce new state for an origin
+>     that persists across browsing sessions?
+
+No.
+
+> 06. Do the features in your specification expose information about the
+>     underlying platform to origins?
+
+Not directly, and not any new information. In general, measuring code execution
+time could potentially be used to determine something about the underlying
+platform, e.g. slow vs. fast machine, when comparing across users.
+`postTask()` adds another entrypoint for executing JavaScript, but there are
+many others and there is nothing inherently different about `postTask()` in
+that sense.
+
+> 07. Does this specification allow an origin to send data to the underlying
+>     platform?
+
+No.
+
+> 08. Do features in this specification allow an origin access to sensors on a user’s
+>     device?
+
+No.
+
+> 09. What data do the features in this specification expose to an origin? Please
+>     also document what data is identical to data exposed by other features, in the
+>     same or different contexts.
+
+The features do not expose any data directly to origins, but we consider
+potential information leaks from timing-based side-channel attacks below.
+
+**Consideration 1**: Can delayed tasks be used to implement a high-resolution timer?
+
+This seems unlikely. `postTask()`'s delay, like `setTimeout()'s`, is expressed
+in whole milliseconds (the minimum non-zero delay being 1 ms), and there is no
+guarantee that tasks will run exactly when the delay expires since tasks are
+queued when the delay expires.
+
+**Consideration 2**:  Does `postTask()` leak information about other origins' tasks?
+
+The threat model we consider is code from two different origins running in
+separate event loops (the same event loop case is Q1). Attackers could attempt
```

Comment:
I think this parenthetical is no longer good.

Reply:
Ugh, I meant to remove that. Thanks for catching that!